GitUI/CommandsDialogs/RevisionDiffControl.cs

```diff
@@ -1443,20 +1391,21 @@ private bool ResetSelectedFilesWithConfirmation()
             }
 
             // Reset to first (parent)
-            ResetSelectedItemsWithConfirmation(resetToSelected: false);
+            ResetSelectedItemsWithConfirmation(resetToParent: true);
             return true;
         }
 
-        private void ResetSelectedItemsWithConfirmation(bool resetToSelected)
+        private void ResetSelectedItemsWithConfirmation(bool resetToParent)
         {
             IEnumerable<FileStatusItem> items = DiffFiles.SelectedItems;
 
+            // new could be changed when restting, allow user to add to checkbox
+            bool hasNewFiles = true;
```

Comment:
This is unlucky. At least for a single file selection, it should be set correctly.

![image](https://github.com/gitextensions/gitextensions/assets/36601201/4c5d35be-25ea-4888-8b85-16dffce905cc)

Reply:
You do not know if a file will be considered new after a checkout, like for renamed files.
This enables the checkbox, do not set it by default.